batavia/types/StrUtils.js

```diff
@@ -584,7 +583,7 @@ function _substitute(format, args) {
 
                 case ('r'):
                     if (types.isinstance(conversionArgRaw, types.Str)) {
-                        conversionArg = `'${conversionArgValue}'`
+                        conversionArg = String(conversionArgValue)
```

Comment:
This actually caused a regression on the 3.6 tests. The output went from `'string'`  to `string`.

Reply:
%r should be format with repr() and %s below should be format with str() so both the 'r' and 's' cases here could be replaced with:

```
                case ('r'):
                    conversionArg = callables.call_method(conversionArgRaw, '__repr__')
                    break
                case ('s'):
                    conversionArg = callables.call_method(conversionArgRaw, '__str__')
                    break
```

Reply:
Or, even better because it will handle JS types as well as Python:

```
                case ('r'):
                    var builtins = require('../builtins')
                    conversionArg = builtins.repr([rawValue], {})
                    break
                case ('s'):
                    var builtins = require('../builtins')
                    conversionArg = builtins.str([rawValue], {})
                    break
```